src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregatingTagger.cs

```diff
@@ -38,6 +42,30 @@ private class AggregatingTagger : ForegroundThreadAffinitizedObject, IAccurateTa
                 // we have an underlying tagger responsible for asynchrounously handling diagnostics
                 // from the owner of that diagnostic update.
                 _owner._diagnosticService.DiagnosticsUpdated += OnDiagnosticsUpdated;
+
+                // Kick off a background task to collect the initial set of diagnostics.
+                var cancellationToken = _initialDiagnosticsCancellationSource.Token;
+                var asyncToken = _owner._listener.BeginAsyncOperation(GetType() + ".GetInitialDiagnostics");
+                var task = Task.Run(() => GetInitialDiagnostics(cancellationToken), cancellationToken);
+                task.CompletesAsyncOperation(asyncToken);
```

Comment:
Is there any concern that the ```OnDiagnosticsUpdated(...)``` might be triggered before the task completes? Or, would the effect essentially be a wash (i.e. the same diagnostics)?